Tests/Opc.Ua.Core.Tests/Types/Encoders/JsonEncoderBenchmarks.cs

```diff
@@ -331,4 +335,764 @@ public void GlobalCleanup()
+            {
+                EscapeString(m_testString);
+            }
+            m_streamWriter.Flush();
+        }
+
+        /// <summary>
+        /// A new implementation using ThreadLocal StringBuilder.
+        /// </summary>
+        [Benchmark]
+        public void EscapeStringThreadLocal()
+        {
+            m_memoryStream.Position = 0;
+            int repeats = InnerLoops;
+            while (repeats-- > 0)
+            {
+                EscapeStringThreadLocal(m_testString);
+            }
+            m_streamWriter.Flush();
+        }
+
+        /// <summary>
+        /// A new implementation using ReadOnlySpan.
+        /// </summary>
+        [Benchmark]
+        public void EscapeStringSpan()
+        {
+            m_memoryStream.Position = 0;
+            int repeats = InnerLoops;
+            while (repeats-- > 0)
+            {
+                EscapeStringSpan(m_testString);
+            }
+            m_streamWriter.Flush();
+        }
+
+        /// <summary>
+        /// A new implementation using ReadOnlySpan and char write.
+        /// </summary>
+        [Benchmark]
+        public void EscapeStringSpanChars()
+        {
+            m_memoryStream.Position = 0;
+            int repeats = InnerLoops;
+            while (repeats-- > 0)
+            {
+                EscapeStringSpanChars(m_testString);
+            }
+            m_streamWriter.Flush();
+        }
+
+        /// <summary>
+        /// A new implementation using ReadOnlySpan and char write.
+        /// </summary>
+        [Benchmark]
+        public void EscapeStringSpanCharsInline()
+        {
+            m_memoryStream.Position = 0;
+            int repeats = InnerLoops;
+            while (repeats-- > 0)
+            {
+                EscapeStringSpanCharsInline(m_testString);
+            }
+            m_streamWriter.Flush();
+        }
+
+        /// <summary>
+        /// A new implementation using ReadOnlySpan and IndexOf.
+        /// </summary>
+        [Benchmark]
+        public void EscapeStringSpanIndex()
+        {
+            m_memoryStream.Position = 0;
+            int repeats = InnerLoops;
+            while (repeats-- > 0)
+            {
+                EscapeStringSpanIndex(m_testString);
+            }
+            m_streamWriter.Flush();
+        }
+
+        /// <summary>
+        /// A new implementation using ReadOnlySpan and Dictionary.
+        /// </summary>
+        [Benchmark]
+        public void EscapeStringSpanDict()
+        {
+            m_memoryStream.Position = 0;
+            int repeats = InnerLoops;
+            while (repeats-- > 0)
+            {
+                EscapeStringSpanDict(m_testString);
+            }
+            m_streamWriter.Flush();
+        }
+
+        [Theory]
+        [TestCase("No Escape chars", 0)]
+        [TestCase("control chars escaped", 1)]
+        [TestCase("binary chars escaped", 2)]
+        [TestCase("mixed escape chars", 3)]
```

Comment:
new test cases were added, forgot to add them

Reply:
updated